### PR TITLE
Fix 340

### LINKED ISSUE #340: Tests for weekplan screen are not testing the screen but is a copy of the tests for the bloc
**Is your feature request related to a problem? Please describe.**
Somebody just copy pasted the tests from the weekplan bloc into the test file for the weekplan screen. Linje 187 - 664 i weekplan_screen_test og linje 91 - 536 i weekplan_bloc_test is identical tests. 

**Describe the solution you'd like**
The bloc_test in the weekplan_screen_test should be deleted and widget tests for the weekplan_screen should be made. 

**Describe alternatives you've considered**
None

**Additional context**
In issue #317 we discovered this problem, when writing some tests, also when trying to write the tests we figured out a problem when trying to pump a widget instance of WeekplanScreen it seems to not create the correct icons in the GirafAppBar, because it seems like the mode it reads from the Authenticator Bloc is incorrect. This meant that we could not find the edit button, which meant we could not test our issue.

### PR DESCRIPTION
**Pull request by EsbenNedergaard**
Fixes #340 

I would recommend not looking at this through GitHub as we for an example had to remove alot of code from weekplan screen test, and then add some new code. So i would instead just see the entire file as new. 

Also we did a lot of moving of code away from the weekplan screen, because we needed some new widget types for testing. 
 - ActivityCard (is just code moved away from weekplan screen)
 - WeekplanDayColumn (also just code move away from weekplan screen)

### CODE REVIEW

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw612f20 you have been assigned this. Here's your checklist.
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [x] The code does it introduce functionality that is not necessary for solving the problem

**Code Readability and Maintainability**

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [x] All new functionality is covered by tests
- [x] It is understandable, by reading the tests, what they do
- [x] The tests covers sensible cases. Both happy paths and exception paths
- [x] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [x] Error messages in the code are understandable
- [x] Confusing sections of code are covered by comments

**Functionality**

For this part, open the app, and test:

- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

**Points for Discussion and Reflection**

- [x] UI messages, that the user might meet, are actually helpful
- [x] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [x] There is well written code in this. Remember to tell the developer!

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw615f20 you have been assigned this. Here's your checklist.
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**

- [ ] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [ ] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [ ] The code does it introduce functionality that is not necessary for solving the problem

**Code Readability and Maintainability**

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [ ] All new functionality is covered by tests
- [ ] It is understandable, by reading the tests, what they do
- [ ] The tests covers sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [ ] Error messages in the code are understandable
- [ ] Confusing sections of code are covered by comments

**Functionality**

For this part, open the app, and test:

- [ ] The features that the code claims to implement, are actually implemented
- [ ] The code does not look like it has bugs
- [ ] All the new screens are reachable through in-app navigation

**Points for Discussion and Reflection**

- [ ] UI messages, that the user might meet, are actually helpful
- [ ] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [ ] There is well written code in this. Remember to tell the developer!

File: test/mock_data.dart [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  }
+
+  @override
+  Observable<SettingsModel> updateSettings(String id, SettingsModel settings) {
+    _mockSettings = settings;
+    return Observable<SettingsModel>.just(_mockSettings);
+  }
+}
+
+// TODO(eneder17): få tjekket vi ikke bruger add() nogle steder
```
Comment (aerlingsson) [inline test/mock_data.dart]: Er den her TODO lavet?

File: pubspec.lock [hunk 2]
```diff
@@ -391,4 +405,4 @@ packages:
     version: "3.5.0"
 sdks:
   dart: ">=2.7.0 <3.0.0"
-  flutter: ">=1.12.13 <2.0.0"
+  flutter: ">=1.12.13+hotfix.5 <2.0.0"
```
Comment (aerlingsson) [inline pubspec.lock]: Er de her version opdateringer nødvendige og har I tjekket at det ikke skaber konflikt på develop?

**Review by aerlingsson — CHANGES_REQUESTED** [review]

**Review by jbjarn17 — CHANGES_REQUESTED** [review]
I can't add an activity/pictogram to a week plan on your branch.
To be sure I went to the develop branch, where I was able to do it.

File: pubspec.lock (hunk 2 above)
Comment (EsbenNedergaard) [reply pubspec.lock]: Tror det er et uheld at det er pushet ændrer det lige tilbage

File: test/mock_data.dart (hunk 1 above)
Comment (EsbenNedergaard) [reply test/mock_data.dart]: Tror ikke vi bruger den nogle steder, men har lige mocket den aligevel for en sikkerheds skyld

**Comment by EsbenNedergaard** [comment]
> I can't add an activity/pictogram to a week plan on your branch.
> To be sure I went to the develop branch, where I was able to do it.

I seem to have fixed it now

**Review by jbjarn17 — APPROVED** [review]

**Review by aerlingsson — APPROVED** [review]

**Pull request merged**